libraries/chain/block_log.cpp

```diff
@@ -453,13 +675,20 @@ namespace eosio { namespace chain {
       if (my->block_file.tellp() <= sizeof(pos))
          return {};
 
+      //figure out if this is a pruned log or not. we can't just look at the configuration since
+      // read_head() is called early on, and this isn't hot enough to warrant a member bool to track it
+      my->block_file.seek(0);
+      uint32_t current_version;
+      fc::raw::unpack(my->block_file, current_version);
+      bool is_currently_pruned = detail::is_pruned_log_and_mask_version(current_version);
+
       my->block_file.seek_end(-sizeof(pos));
-      my->block_file.read((char*)&pos, sizeof(pos));
-      if (pos != npos) {
+      if(is_currently_pruned)
```

Comment:
I reordered the seeks & skips to be more in line with what is actually be skipped over in the file. I don't think the mouthful with the ternary operator improves readability and the extra seek isn't much of a big deal for this function since it isn't hot.